data-prepper-core/src/main/java/org/opensearch/dataprepper/sourcecoordination/LeaseBasedSourceCoordinator.java

```diff
@@ -309,40 +308,46 @@ public void updatePartitionForAcknowledgmentWait(final String partitionKey, fina
         itemToUpdate.setPartitionOwnershipTimeout(Instant.now().plus(ackowledgmentTimeout));
 
         sourceCoordinationStore.tryUpdateSourcePartitionItem(itemToUpdate);
-
-        partitionManager.removeActivePartition();
     }
 
     @Override
     public void giveUpPartitions() {
-
         if (!initialized) {
             return;
         }
-
         final Optional<SourcePartition<T>> activePartition = partitionManager.getActivePartition();
```

Comment:
This won't have a partition since you changed the code to not track partition manager. There are two options here

1. Have partition manager track all the partitions that threads are working on, and call giveUpPartition on each
2. Remove the `PartitionManager` class completely, and then on shutdown have threads stop and give up their individual partition

I would say option 2 is easier